Some updates :)

HideakiAtsuyo opened the pull request.

HideakiAtsuyo commented:
Need:
- Create dev branch
- Set dev bool to true in Client.cpp(for dev branch)

luccanunes commented on README.md:
```diff
@@ -12,27 +12,58 @@ Inspired by the simplicity of [discord.py](https://github.com/Rapptz/discord.py)
- client.run(token);
-}
+    const std::vector<std::string> owners = { "ID1", "ID2", "..." }; //You can add some IDs(Bot Owners only) !
+    const std::string prefix = ">"; //Bot Prefix
+    
+    const std::string token = "Bot token";
+
+    dpp::Client client;
+
+    client.onReady = [&]() {
+        std::cout << client.user.name << " ready!\n\n";
+    };
+
+
+    client.onMessage = [&](dpp::Message message) {
+
+        if (message.author.id == client.user.id) return;
+        //if (message.author.bot) return message.channel.send("Bots aren't allowed"); //message.author.bot doesn't return user for the moment :/
+        //Waiting the `startWith` to add return if the prefix isn't in the message and more..
+        if (message.content == prefix + "ownersonly") { //Prefix with command
+            if (std::find(std::begin(owners), std::end(owners), message.author.id) != std::end(owners)) { //Little example for owners only(will be updated later)
+                message.reply("Allowed");
+                return;
+            }
+            else {
+                message.reply("Not Allowed");
+                return;
+            };
+
+        }
+        else if (message.content == prefix + "help") { //Prefix with command
+            message.reply("In progress..");
+        }
+        else {
+            //message.reply("command"); //Need to wait for startWith to avoid a response without prefix!
+        };
+    };
+    client.run(token);
+};
+
```
Although I like what you went for here, I think we should keep the REAME.md example as simple as we can, and includen this in some documentation later.

HideakiAtsuyo replied:
I leave it like that for those who want some more advanced examples but you can modify it if you want to

luccanunes replied:
I'll add such examples in the documentation website

luccanunes commented on src/Api.cpp:
```diff
@@ -1,41 +1,108 @@
+					{"d", lastS}
+				};
+				if(dev){ std::cout << "SENDING HEARTBEAT: " << payload.dump() << "\n\n"; }
+				webSocket.send(payload.dump());
+			}
+			std::this_thread::sleep_for(std::chrono::milliseconds(200));
+		}
 	}
-	json put(const std::string& path) {
-		const std::string URL = baseURL + path;
-		std::cout << "URL: " << URL << std::endl;
-		cpr::Response res = cpr::Put(
-			cpr::Url{ URL },
-			cpr::Header{ {"authorization", auth}, {"Content-length", "0"} }
-		);
-		std::cout << "Response: " << res.text << std::endl;
-		json result;
-		if (res.text.length() != 0)
-			result = json::parse(res.text);
-		else
-			result = json::parse("{}");
-		return result;
+	void Client::sendID() {
+		json props = { {"$os", "linux"}, {"$browser", "discord_cpp"}, {"$device", "discord_cpp"} };
+		json id = {
+			{"op", 2},
+			{"d", {
+				{"token", token}, {"properties", props}, {"intents", 513}
+			}}
+		};
+		if(dev){ std::cout << "SENDING: " << id.dump() << "\n\n"; }
+		webSocket.send(id.dump());
+	}
+	void Client::send(const std::string message, const std::string channel_id) {
+		const std::string path = "/channels/" + channel_id + "/messages";
+		json body = { {"content", message} };
+		json res = Api::post(path, cpr::Body{ body.dump() });
+		if(dev){ std::cout << res << std::endl; }
 	}
 }
```
Don't fully understand what's going on here. Did you mean to add this changes to Client.cpp instead of Api.cpp?

luccanunes replied:
No worries

luccanunes commented on src/Message.cpp:
```diff
@@ -43,4 +43,7 @@ namespace dpp {
 	}
-}
\ No newline at end of file
+	bool Message::startsWith(const std::string& s) {
+		return content.find(s) == 0;
+	}
```
This feature has already been added in recent commits at the `main` branch

HideakiAtsuyo replied:
I know i just do that to be sure that everything is correct

HideakiAtsuyo commented:
Checked all your commits and updated everything in the pull request

HideakiAtsuyo commented:
Checked all your commits and updated everything in the pull request

luccanunes commented:
I'm sorry, but I'll have to close this one. There are just a lot of conflicts and not really anything new.
I've already removed all logging from the `main` branch and added it to the `development` one.
If you can, please refrain from making such large pull requests, try to split different topics into different PRs.
Thanks for contributing once again!

HideakiAtsuyo commented:
> 
> 
> I'm sorry, but I'll have to close this one. There are just a lot of conflicts and not really anything new.
> I've already removed all logging from the `main` branch and added it to the `development` one.
> If you can, please refrain from making such large pull requests, try to split different topics into different PRs.
> Thanks for contributing once again!

the dev bool is better than just remove the std::cout because like this everyone can use it but ok

luccanunes commented:
I get what you mean but all those logs aren't really necessary to the final user. Specially since all methods (like add_reaction) return a string in json format, with the API response
